classes/manager.php

```diff
@@ -147,8 +147,12 @@ public static function is_factorid_valid($factorid, $user) {
      * @return void
      */
     public static function cannot_login() {
+        global $OUTPUT;
+        echo $OUTPUT->header();
+        $OUTPUT->not_enough_factors();
```

Comment:
OUTPUT methods should return html not echo them, so you can buffer and compose them etc